dvc/remote/http.py

```diff
@@ -99,3 +99,9 @@ def _request(self, method, url, **kwargs):
 
     def gc(self):
         raise NotImplementedError
+
+    def all(self):
+        raise NotImplementedError(
```

Comment:
But how is config option related to the `Remote.all`? :slightly_smiling_face: Also, `all` is implemented in RemoteBASE, it is list_cache_paths that is not implemented in RemoteHTTP and would make more sense to overwrite this way than `all()`. Also, NotImplementedError would show up as an `unexpected error`, which shouldn't be that way for preventable errors. That is why it would be better to use RemoteActionNotImplemented as we do in other places. That being said, this is more of a ConfigError, isn't it? And it should be raised in __init__, so no one is able to use a misconfigured RemoteHTTP instance.

Reply:
> But how is config option related to the Remote.all?
> Also, all is implemented in RemoteBASE

Indeed, didn't notice `remote.all` was already implemented.

> That is why it would be better to use RemoteActionNotImplemented as we do in other places.

ohh, didn't now about the `RemoteActionNotImplemented`.

> That being said, this is more of a ConfigError, isn't it? And it should be raised in RemoteHTTP.__init__, so no one even has a chance to use a misconfigured RemoteHTTP instance.

Well, yes, but it is _misconfigured_ only for certain operations , I'd rather leave it under `list_cache_paths`

Reply:
@mroutis Ok, so now with your current patch when user will try to run status/push/pull/etc for a large project with millions of files, he will wait for a few minutes until it checks everything locally and only then will be greeted with this error, right? That sounds bad. Also, you've overwritten a method from RemoteBASE that is already implementing NotImportedError. RemoteHTTP is parsing config for that option and the quickest and the earliest time where we can throw the error is `RemoteHTTP.__init__`.

Reply:
Why `RemoteBase` is using `NotImplementedError` instead of the `RemoteActionNotImplemented` ?

I'll move it to `RemoteHTTP.__init__`

Reply:
You mean `RemoteBASE.list_cache_paths`? Looks like we've either missed it or it didn't exist when we wrote those lines. I see there are a bunch of such places in RemoteBASE, we could fix that in a separate PR :slightly_smiling_face: